### PR TITLE
Add explanation for workaround of #3346 

### PR DESCRIPTION
**Pull request by wucas**
Adds an explanation for a workaround for #3346 to INSTALL.md. This way it might be easier to find an explanation if others have the same issue.

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -491,15 +491,15 @@ finite) which are required for the available algorithms. See section
...
 manual.
 
-Problems specific to Windows
-
 ### The ^-key or "-key cannot be entered.
 
 This is a problem if you are running a keyboard driver for some non-english
 languages. These drivers catch the ^ character to produce the French
-circumflex accent and do not pass it properly to GAP. No fix is known. (One
+circumflex accent and do not pass it properly to GAP. For OS X users, as a workaround please refer to the section "GAP for OS X" below for information on how to install readline and section 5 on how to recompile GAP, for windows no fix is known. (One
```
Comment (fingolfin) [inline INSTALL.md]: Please word wrap this line to ~80 characters.

File: INSTALL.md [hunk 2]
```diff
@@ -491,15 +491,15 @@ finite) which are required for the available algorithms. See section
...
 can type POW(a,b) for a^b.)
 
+Problems specific to Windows
```
Comment (fingolfin) [inline INSTALL.md]: This is not your fault, but this line is really weird. Was this perhaps meant to be a section headline? If so, perhaps it should be `## Problems ...` ? Perhaps @alex-konovalov knows?

File: INSTALL.md [hunk 3]
```diff
@@ -609,6 +609,32 @@ a terminal in order to make all required tools available via the command line:
 
      xcode-select --install
 
+You might want to consider using Readline by installing it via:
```
Comment (fingolfin) [inline INSTALL.md]:
```suggestion
You might want to consider using GNU readline by installing it via:
```
There are several readline implementation; macOS (more generally: BSD based systems) has its own `libreadline`, so let's be specific here.

File: INSTALL.md [hunk 4]
```diff
@@ -609,6 +609,32 @@ a terminal in order to make all required tools available via the command line:
...
+
+ * using Homebrew: `brew install readline`
+ * using Fink: `fink install readline7`
+ * using MacPorts: `port install readline`
+ 
+After that you have to compile GAP and tell it where to find your installation of Readline
```
Comment (fingolfin) [inline INSTALL.md]:
```suggestion
After that you have to compile GAP and tell it where to find your installation of GNU readline
```
Also, please remember to word wrap at ~80 chars.

File: INSTALL.md [hunk 5]
```diff
@@ -609,6 +609,32 @@ a terminal in order to make all required tools available via the command line:
...
+using the following commands.
+
+For Homebrew, use these commands:
+
+    ./configure --with-readline=/usr/local/opt/readline
```
Comment (fingolfin) [inline INSTALL.md]:
```suggestion
    ./configure --with-readline=$(brew --prefix)/opt/readline
```
covers more users

File: INSTALL.md [hunk 6]
```diff
@@ -609,6 +609,32 @@ a terminal in order to make all required tools available via the command line:
...
+
+    ./configure CPPFLAGS=-I/sw/include LDFLAGS=-L/sw/lib
+    make
+
+For MacPorts, use these commands:
+
+    ./configure CPPFLAGS=-I/opt/local/include LDFLAGS=-L/opt/local/lib
+    make
+
+For further information on how to use Readline refer to section 5 above.
```
Comment (fingolfin) [inline INSTALL.md]:
```suggestion
For further information on how to use GNU readline, refer to section 5 above.
```

File: INSTALL.md (hunk 2 above)
Comment (wucas) [reply INSTALL.md]: I wondered about this as well.  I can change it if that's okay.
```suggestion
##Problems specific to Windows
```

Comment (fingolfin) [reply INSTALL.md]: There should be a space after `##`

File: INSTALL.md [hunk 7]
```diff
@@ -491,15 +491,18 @@ finite) which are required for the available algorithms. See section
...
-languages. These drivers catch the ^ character to produce the French
-circumflex accent and do not pass it properly to GAP. No fix is known. (One
+languages. These drivers catch the ^ character to produce the French circumflex
+accent and do not pass it properly to GAP. For OS X users, as a workaround
+please refer to the section "GAP for OS X" below for information on
+how to install readline and section 5 on how to recompile GAP, for windows no
+fix is known. (One
 can type POW(a,b) for a^b.)
 
+Problems specific to Windows
```
Comment (fingolfin) [inline INSTALL.md]:
```suggestion
## Problems specific to Windows
```
I quickly looked at how this renders, and I'd suggest to indeed changes this. After this, I am only wondering if the separation into three commits here really is warranted, perhaps you could briefly explain it, or alternatively, squash it down. But that's a minor concern, in principle I'd be fine with this to be merged :-). Would be good to hear if @alex-konovalov has any further input, though.

**Comment by wucas** [comment]
> Thanks! Some minor remarks attached. Is this in three commits on purpose?

No not on purpose I just was not aware of the right workflows, I squashed them.

**Comment by wucas** [comment]
@alex-konovalov I've also changed OS X to macOS. For consistency I changed it in the README.md too.

**Comment by fingolfin** [comment]
@wucas minor thing: your commits show up with different names. One has author `Lucas Wollenhaupt <reg-github@wucas.de>` and one has `Lucas Wollenhaupt <github@wucas.de>` Is that intentional? If you want to use both addresses, I recommend to add both to your GitHub profile, so that GitHub can correctly associate your commits to your GitHub account.

File: INSTALL.md [hunk 8]
```diff
@@ -491,15 +491,18 @@ finite) which are required for the available algorithms. See section
...
 
 This is a problem if you are running a keyboard driver for some non-english
-languages. These drivers catch the ^ character to produce the French
-circumflex accent and do not pass it properly to GAP. No fix is known. (One
+languages. These drivers catch the ^ character to produce the French circumflex
+accent and do not pass it properly to GAP. For macOS users, as a workaround
+please refer to the section "GAP for macOS" below for information on
+how to install readline and section 5 on how to recompile GAP, for windows no
+fix is known. (One
 can type POW(a,b) for a^b.)
```
Comment (fingolfin) [inline INSTALL.md]: Maybe move that up to the preceding line? It looks a bit weird this way.

File: INSTALL.md [hunk 9]
```diff
@@ -582,11 +585,11 @@ debug mode enabled:
     ./configure CC=clang-5.0 CXX=clang++-5.0 CFLAGS="-g -Og" CXXFLAGS="-g -Og" --enable-debug
 
 
-13 GAP for OS X
+13 GAP for macOS
 ===============
```
Comment (fingolfin) [inline INSTALL.md]:
```suggestion
================
```

File: INSTALL.md [hunk 10]
```diff
@@ -94,10 +94,10 @@ for GAP and some packages and provides the standard installation procedure.
 The exact method of unpacking will vary dependently on the operating system
 and the type of archive used.
 
-Unix (including OS X)
+Unix (including macOS)
 ---------------------
```
Comment (fingolfin) [inline INSTALL.md]:
```suggestion
----------------------
```

**Pull request merged**